src/models/basemodel.h

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BASEMODEL_H
+#define BASEMODEL_H
+
+#include <QByteArray>
+
+class BaseModel {
+ public:
+  BaseModel() = default;
+  virtual ~BaseModel() = default;
+
+  bool initialized() const { return !m_rawJson.isEmpty(); }
+
+  void forget() {
+    m_rawJson.clear();
+    m_writeNeeded = false;
+  }
+
+ protected:
+  void storeJson(const QByteArray& json, bool setWriteNeeded) {
+    m_rawJson = json;
+    if (setWriteNeeded) {
+      m_writeNeeded = true;
+    }
+  }
+
+  bool compareJson(const QByteArray& json) const {
+    return json.isEmpty() || m_rawJson != json;
```

Comment:
I am slightly confused as to why `json.isEmpty()` is treated specially here. Surely if `m_rawJson` is empty, and it stays empty, then there's no reason to write anything.